astropy/table/table.py

```diff
@@ -1069,7 +1069,11 @@ def __array__(self, dtype=None):
         Coercion to a different dtype via np.array(table, dtype) is not
         supported and will raise a ValueError.
         """
-        if dtype is not None:
+        if np.dtype(dtype).kind == 'O':
```

Comment:
What is the intent here? If `dtype` is `None`, this becomes `float64`. Is that correct?

Reply:
No, then `dtype` of `array` is inferred from the input, i.e. unchanged from the table dtypes.

Reply:
But then it will fall through, since `np.dtype(None).kind != 'O'` (see updated suggestion), so actually does work as intended.